src/common/input/config.rs

```diff
@@ -44,6 +47,35 @@ impl Default for Config {
     }
 }
 
+impl TryFrom<&CliArgs> for Config {
+    type Error = ConfigError;
+
+    fn try_from(opts: &CliArgs) -> ConfigResult {
+        if let Some(ref path) = opts.config {
+            return Config::new(&path);
+        }
+
+        if let Some(ConfigCli::Config { clean, .. }) = opts.option {
+            if clean {
+                return Config::from_default_assets();
+            }
+        }
+
+        let config_dir = opts.config_dir.clone().or_else(install::default_config_dir);
```

Comment:
I misunderstood the behavior of `default_config_dir`, which I thought would create a directory.
`default_config_dir` just indicates a possible configuration directory, so it should be called here as the previous implementation.

By the way, as `default_config_dir` returns `Option<PathBuf>`, I made `opts.config_dir` call `clone()` to match these types. Maybe, I can delete this `clone()` by refactoring `default_config_dir`, but it should be done in another PR.

Reply:
>I misunderstood the behavior of `default_config_dir`, which I thought would create a directory.
`default_config_dir` just indicates a possible configuration directory, so it should be called here as the previous implementation.

Yeah, I glossed over that too now. The naming does suggest some sort of creation. I'll put it on my list to try to make the intention clearer.

>By the way, as `default_config_dir` returns `Option<PathBuf>`, I made `opts.config_dir` call `clone()` to match these types. Maybe, I can delete this `clone()` by refactoring` default_config_dir`, but it should be done in another PR.

I would love that, as I am also unhappy about the state of `default_config_dir`, so feel free to open a PR on that, if you want to!